src/freertos/CMakeLists.txt

```diff
@@ -33,8 +32,17 @@ target_include_directories(${LIB_NAME}
 
 target_link_libraries(${LIB_NAME}
     PUBLIC
+    FreeRTOS::Coroutine
+    FreeRTOS::EventGroups
+    FreeRTOS::StreamBuffer
         FreeRTOS::Timers
-        FreeRTOS::Heap::1
+        FreeRTOS::Heap::4
```

Comment:
Quick question: in FreeRTOS::Heap::X, what does the number stands for?

Reply:
The heap implementation, heap 1 is "static" so freertos has a large buffer and manages it, allows heap on devices that doesn't have one. Heap2 is like heap1 but with some logic to avoir framentation. heap3 is simply a wrapper over malloc and free, so it uses the device heap. Heap4 is like 2 but better at the cost of some memory overhead. Heap5 is like heap4 but I think you can have multiple non-contiguous heap spaces.   We may change to heap3 eventually, or reduce our platform heap to 0. I'm not sure which we should use yet. You can have more info here: https://www.freertos.org/a00111.html

Reply:
I think we will go for heap3, heap4 has the issue that if we use it on posix, the size_t and the architecture is bigger, thus takes more heap for allocation. In other words, a small heap that works on stm32 doesn't mean it works for posix. We may eventually just encapsulate it in the freertos platform if it becomes an issue